test/e2e.method.signing.js

```diff
@@ -173,6 +184,89 @@ describe('transaction and message signing [ @E2E ]', function() {
         }
     });
 
+    it('accounts.signTransaction errors when tx signing is invalid', async function(){
+        const source = wallet[0].address;
+        const destination = wallet[1].address;
+
+        const txCount = await web3.eth.getTransactionCount(source);
+
+        // Using gas === 0 / ethereumjs-tx checks this wrt common baseFee
+        const txObject = {
+            nonce:    web3.utils.toHex(txCount),
+            to:       destination,
+            value:    web3.utils.toHex(web3.utils.toWei('0.1', 'ether')),
+            gasLimit: web3.utils.toHex(0),
+            gasPrice: web3.utils.toHex(web3.utils.toWei('10', 'gwei')),
+            hardfork: "istanbul",
+            chain:    "ropsten",
+        };
+
+        try {
+            await web3.eth.accounts.signTransaction(txObject, wallet[0].privateKey);
+            assert.fail()
+        } catch (err) {
+            assert(err.message.includes('gas limit is too low'));
+        }
+    })
+
+    it('accounts.signTransaction errors when no transaction is passed', async function(){
+        try {
+            await web3.eth.accounts.signTransaction(undefined, wallet[0].privateKey);
+            assert.fail()
+        } catch (err) {
+            assert(err.message.includes('No transaction object'));
+        }
+    });
+
+    it('transactions sent with wallet error correctly (with receipt)', async function(){
+        var data = instance
+            .methods
+            .reverts()
+            .encodeABI();
+
+        var tx = {
+            from: wallet[0],
+            to: instance.options.address,
+            data: data,
+            gasPrice: '1',
+            gas: 4000000
+        }
+
+        try {
+            await web3.eth.sendTransaction(tx);
+            assert.fail();
+
```

Comment:
```suggestion
```

Reply:
Fixed in 269d11943c0db30f11effb227856ecb473bec07c